library/src/main/java/org/apache/apex/malhar/lib/state/managed/AbstractManagedStateImpl.java

```diff
@@ -128,14 +129,14 @@
 {
   private long maxMemorySize;
 
-  protected int numBuckets;
+  protected long numBuckets;
```

Comment:
@siyuanh  Sorry there is something wrong with my github email notification so I didn't know you had changed #503. I have replied to your comments about time buckets for session windows there.